force-app/main/default/classes/CarTriggerHelper.cls

```diff
@@ -0,0 +1,37 @@
+public with sharing class CarTriggerHelper {
+    
+    public static void updateCar(Map<Id,Product2> oldCars){
+        
+        List<Product2> updatedCars=[SELECT Id,Car_Status__c,Condition__c,Name,reservedInOpportunity__c FROM Product2 WHERE Id IN:Trigger.new];
+        
+        Set<Id> findCarsId=new Set<Id>();
+        List<Opportunity> opportunityToUpdate=new List<Opportunity>();
+        
+        for(Product2 car:updatedCars){
+            Product2 oldCar=oldCars.get(car.Id);
+            
+            if(oldCar!=null){
```

Comment:
You run this method on update only, it will always have oldCar record, so you can remove this if statement